src/Cardano/Prelude/HeapWords.hs

```diff
@@ -453,3 +458,51 @@ instance HeapWords e => HeapWords (V.Vector e) where
 
 heapWordsUVector :: V.U.Unbox e => Int -> V.U.Vector e -> Int
 heapWordsUVector sz a = 5 + (V.U.length a * sz) `div` wordSize
+
+instance HeapWords Natural where
+  heapWords (NatS# _)
+    -- We have
+    --
+    -- > NatS# GmpLimb#
+    -- > type GmpLimb# = Word#
+    --
+    -- so @(NatS# n)@ requires:
+    --
+    -- - 1 word for the header 'NatS#' object header
+    -- - 1 word for the single 'NatS#' unboxed field, of type 'Word#'
+    --
+    = 1 + 1
+  heapWords (NatJ# (BN# arr))
+    -- We have
+    --
+    -- > NatJ# {-# UNPACK #-} !BigNat
+    -- > data BigNat = BN# ByteArray#
+    --
+    -- so @NatJ# (BN# xs)@ requires:
+    --
+    -- - 1 word for the 'NatJ#' object header
+    -- - 1 word for the 'NatJ#' pointer field of type 'ByteArray#'
+    -- - the words used by the byte array.
+    --
+    = 1 + 1 + heapWordsByteArray# arr
```

Comment:
I think we should break this down using a `HeapWords BigNat` instance as we'll want that for `HeapWords Integer` as well. So this would then be:
```
heapWords (NatJ# bn) = 1 + 1 + heapWordsUnpacked bn
```
which makes it clear that we unpack the `BigNat`.

Reply:
Done. Please see if this makes sense now.